MongoDB documentation improvements (#8515)
quarkusio/quarkus: docs/src/main/asciidoc/mongodb.adoc

The syntax is as follows: `quarkus.mongodb.[optional name.][mongo connection property]`.
If the name is omitted, it configures the default client.

[NOTE]
====

> Using multiple MongoDB clients allow to implements multi-tenancy for MongoDB by allowing to connect to multiple MongoDB clusters. +

Reviewer: ```suggestion
The use of multiple MongoDB clients enables multi-tenancy for MongoDB by allowing to connect to multiple MongoDB clusters. +
```

If you want to connect to multiple databases inside the same cluster,

> you can use the same client as a client is able to access all databases on the same cluster

Reviewer: ```suggestion
multiple clients are **not** necessary as a single client is able to access all databases in the same cluster
```

link:https://docs.mongodb.com/manual/core/security-client-side-encryption/[Client-Side Field Level Encryption] is also not supported in native mode.
====

[TIP]
====
If you encounter the following error when running your application in native mode: +
`Failed to encode 'MyObject'. Encoding 'myVariable' errored with: Can't find a codec for class org.acme.MyVariable.` +

> This means that the `org.acme.MyVariable` class is not known to GraalVM, to fix this you can add the `@RegisterForReflection` annotation to your `MyVariable` class.

Reviewer: the remedy is to add the `@RegisterForReflection` annotation to your `MyVariable` class.
